libcudacxx/test/libcudacxx/CMakeLists.txt

```diff
@@ -39,6 +39,11 @@ else() # NOT LIBCUDACXX_TEST_WITH_NVRTC
   set(LIBCUDACXX_TEST_COMPILER_FLAGS "-DLIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE")
 endif()
 
+# enable exceptions and assertions in tests
+string(APPEND LIBCUDACXX_TEST_COMPILER_FLAGS
+       " -DLIBCUDACXX_ENABLE_EXCEPTIONS"
```

Comment:
Enabling exceptions seems unrelated to this PR? Why was it added? Oversight?

Reply:
I want to enable assertions globally, so we can just as well enable exceptions too

Reply:
I can imagine this introducing all kinds of new side effects and possible requiring additional fixes unrelated to assertions. Can we please do it in a separate PR?